src/util.c

```diff
@@ -801,6 +801,31 @@ double git_time_monotonic(void)
 	return git__timer();
 }
 
+/**
+ * Iterate through an UTF-8 string and stops after finding any invalid UTF-8
```

Comment:
Sorry, I think you misunderstood my initial comment on this documentation. It should definitely be part of the header, but instead of having a `/** */` header, we want it to be a `/* */` header (note the difference of one `*` vs two `**` characters)